homeassistant/components/technove/helpers.py

```diff
@@ -0,0 +1,41 @@
+"""Helpers for TechnoVE."""
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from typing import Any, Concatenate, ParamSpec, TypeVar
+
+from technove import TechnoVEConnectionError, TechnoVEError
+
+from homeassistant.exceptions import HomeAssistantError
+
+from .entity import TechnoVEEntity
+
+_TechnoVEEntityT = TypeVar("_TechnoVEEntityT", bound=TechnoVEEntity)
+_P = ParamSpec("_P")
+
+
+def technove_exception_handler(
+    func: Callable[Concatenate[_TechnoVEEntityT, _P], Coroutine[Any, Any, Any]],
+) -> Callable[Concatenate[_TechnoVEEntityT, _P], Coroutine[Any, Any, None]]:
+    """Decorate TechnoVE calls to handle TechnoVE exceptions.
+
+    A decorator that wraps the passed in function, catches TechnoVE errors,
+    and handles the availability of the device in the data coordinator.
+    """
+
+    async def handler(
+        self: _TechnoVEEntityT, *args: _P.args, **kwargs: _P.kwargs
+    ) -> None:
+        try:
+            await func(self, *args, **kwargs)
+            self.coordinator.async_update_listeners()
+
+        except TechnoVEConnectionError as error:
+            self.coordinator.last_update_success = False
+            self.coordinator.async_update_listeners()
```

Comment:
By the looks of it, this doesn't have to update the state, this will only update the entities, but there is no new state (iirc).

what about just (request an) update the coordinator?

Reply:
I think I did what you suggested but I'm really not sure I understood your feedback properly.

Also, It takes a few seconds after the request to the station to complete before the state is reflected in the API, which means that when a user turns the switch off, they will see it off for a second, then it jumps back to on until it finally goes to off once the device has fully turned it off.
Do you know if this behaviour can be improved?